packages/aws-cdk/test/integ/helpers/cdk.ts

```diff
@@ -117,11 +118,9 @@ export function withMonolithicCfnIncludeCdkApp<A extends TestContext>(block: (co
 
     let success = true;
     try {
-      let module = uberPackage;
-      if (FRAMEWORK_VERSION) {
-        module = `${module}@${FRAMEWORK_VERSION}`;
-      }
-      await fixture.shell(['npm', 'install', 'constructs', module]);
+      const module = FRAMEWORK_VERSION ? `${uberPackage}@${FRAMEWORK_VERSION}` : uberPackage;
+      const constructs = CONSTRUCTS_VERSION ? `constructs@${CONSTRUCTS_VERSION}` : 'constructs';
+      await fixture.shell(['npm', 'install', constructs, module]);
```

Comment:
could we maybe do `npx npm@7.0.0 install <module>` which would automatically install all peer dependencies?

Reply:
That wouldn't be properly intercepted by our "also works with symlinks in the repo" hackery.

Reply:
Honestly -- it might even be better to switch to generating a `package.json` and deferring to regular NPM.

In the dist/release cases we'll generate a `package.json` that looks like:

```
{ 
  "@aws-cdk/core": "1.85.0"
}
```

And in the repo case we'll generate a `package.json` that looks like:

```
{ 
  "@aws-cdk/core": "/path/to/repo/packages/@aws-cdk/core"
}
```

And then we can use NPM7 to take care of `peerDependencies` (i.e. `constructs`).

I ain't against that!

Reply:
Hmm downside of `npx` btw is that it will toss away the installed package after every run. So we'd be downloading and installing `npm@7` for EVERY TEST if we did that.

Perchance not great :)